### ISSUE TITLE
Null ref: T2.GameState_RompingThroughDungeon.OnStoppedBeingCurrentState

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#750]
Auto report. 0.2.1.

Feedback ID: 20201210-PBKGM

```
4:10:23, Frame 1217, LOG »  NETWORK CLIENT: RelayClient connected to server 107.191.57.170:5889
4:10:24, Frame 1223, WARNING »  NETWORK CLIENT: RelayClient disconnected: RemoteConnectionClose


4:10:25, Frame 1292, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.GameState_RompingThroughDungeon.OnStoppedBeingCurrentState ()
T2.Game.AddGameStateToStack (T2.GameState newGameState)
T2.GameState.AddToStack ()
T2.Game.ForceGameStateAsCurrentGameState (T2.GameState gameState)
T2.AlertBox.ForceGameStateStackPriority ()
T2.AlertBox.Update ()

GameStates: TransitioningToDungeonRoom, RompingThroughDungeon
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: SavingGame
Room: 0 / 0
RoomModifiers: None
YouHaveDiedState: Null
SystemPlayer InputDevice: Keyboard
HumanPlayer 1 InputDevice: Keyboard
HumanPlayer 2 InputDevice: None
```

**Comment by ChrisNZL** [comment]
Fixed in 0.2.3. Added null checks.

**Comment by ChrisNZL** [comment]
Reopening. Auto report in 0.2.10a. Feedback ID: 20211115-R6YV9

```
5:13:37, Frame 297795, LOG »  NetworkMessage_DungeonRoomTransitionHelper: Received message.
Desire: 4 » BeginLoadingRoom

5:13:43, Frame 298094, WARNING »  NETWORK CLIENT: RelayClient disconnected: Timeout
5:13:43, Frame 298094, LOG »  Device.OnBecameOffline: Device is now offline.
5:13:46, Frame 298102, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第16个房间：烈火洞窟
5:13:58, Frame 298813, LOG »  Returning to title screen...

5:13:59, Frame 298884, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.GameState_RompingThroughDungeon.OnStoppedBeingCurrentState ()
T2.Game.RemoveGameStateFromStack (T2.GameState gameStateToRemove)
T2.Game.RemoveTopMostGameStateFromStack ()
T2.Game.HardResetBackToTitleScreen ()
T2.GameState_ReturningToTitleScreen.Update ()

GameStates: Disconnected, RompingThroughDungeon
OnlineGameInfo: Device is offline.
Previous server: Lightning-Beijing-01
Previous GameID: QX3SM
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: RevealingRoom
Room: 16 / 16
RoomModifiers: ManyElites (超多精英怪), BoomerangsOnly (回力镖派对), Penniless (家徒四壁)
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
HumanPlayer 2 InputDevice: None
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.3.1a... Added a couple null checks within `GameState_RompingThroughDungeon.OnStoppedBeingCurrentState`.
